File: backend/apps/executions/api/serializers/log.py
from rest_framework import serializers
from apps.executions.models import Log,Execution


class ExecutionLogSerializer(serializers.ModelSerializer):
    class Meta:
        model = Execution
        fields = [
            "status",
            "date_start",
            "date_end",
        ]


class LogSerializer(serializers.ModelSerializer):
    execution_data = ExecutionLogSerializer(read_only=True,source="execution")
    class Meta:
        model = Log
        fields = [
            'id',
            "description",
            "created_at",
            "updated_at",
            "execution",
            "execution_data"
        ]
        read_only_fields = [
            "created_at",
            "updated_at",
            "execution_data"
        ]

File: frontend/src/types/log.ts
export interface Log {
  id: number;
  description: string | null;
  created_at: string;
  updated_at: string;
  execution: number;
  execution_data: { status: string; date_start: string; date_end: string | null };
}

export interface LogCreate {
  execution: number;
  description?: string;
}

export interface LogFilters {
  execution?: string;
  description?: string;
  created_at?: string;
}
